### ISSUE TITLE
A remoting timeout communicating with a job should cause the job to fail

### ISSUE THREAD

**Issue by TravisEz13** [PowerShell/PowerShell#3393]
Steps to reproduce
------------------
1. **Cause the job to go into the WER debugger**
```
$job = Start-Job {
function CrashPowerShell
{
    $classDefinition = @'
    using System;
    using System.Threading;
    # The following code is to simulate a random crash you may get in real code
    public class PowerShellCrashTest {
        public static void CrashPowerShell() {
            Thread thread = new Thread(() => { throw new Exception(); });
            thread.Start();
            Thread.Sleep(TimeSpan.FromSeconds(60)); // This sleep should not complete because the process should crash shortly after the thread starts.
        }
    }
'@
    Add-Type -TypeDefinition $classDefinition
    [PowerShellCrashTest]::CrashPowerShell()
}

# Again, The following call is to simulate a random crash you may get in real code
CrashPowerShell
}
```
2. **Don't let the process out of the debugger (assume you aren't paying attention)**

If a Windows Error Reporting window (“powershell has stopped working”) shows, do not interact with the window. Wait for a couple of minutes, and check the state of the job through the variable $job.

Expected behavior
-----------------
The job will be at the state “Failed”, and the JobStateInfo will contain an exception.

Actual behavior
---------------
The job will report that it reached the state “Completed”, and the JobStateInfo will not have any exception.

Discussion
-----------

This introduces two issues:
1. We do not detect the crash so that our engine that starts PowerShell jobs during deployment can report it as an error.
2. Even if we work around the first issue (e.g. by adding a line that returns a value to the end of the job to find if execution reached the end), we still do not have a way to get the details of the exception that caused the crash. This makes it very difficult to investigate the underlying root cause of the failure in the script that was running as a job.

Environment data
----------------

```powershell
> $PSVersionTable

Name                           Value
----                           -----
PSVersion                      5.1.14393.953
PSEdition                      Desktop
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0...}
BuildVersion                   10.0.14393.953
CLRVersion                     4.0.30319.42000
WSManStackVersion              3.0
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
```

**Comment by PaulHigin** [comment]
Once the target process crashes the job client probably doesn't get anymore information through the remoting system and the client side is eventually closed due to an inactivity timeout.  This should generate an error on the client side (session broken) that the job should report with a "Failed" status.

**Comment by PaulHigin** [comment]
I found the place were the operation timeout error is being dropped by the job class during error handling.  It is a simple one-line fix and I'll create a PR for it.

**Comment by PaulHigin** [comment]
This was fixed with PR #3445.
